src/amp-story-ad-preview.js

```diff
@@ -76,17 +82,43 @@ export default class AmpStoryAdPreview {
     this.win = win;
     this.element = element;
 
+    /** @private {boolean} */
+    this.useSourcedoc_ = true;
+
+    /**
+     * Controls which method is used to render. Defaults to srcdoc, but
+     * is set to doc write when srcdoc is unsupported.
+     * @private {function}
+     */
+    this.renderer_ = this.renderWithSrcdoc_;
+
+    const {promise, resolve} = new Deferred();
+    /** @private {!function} */
+    this.adIframeResolver_ = resolve;
+    /** @private {!Promise} */
+    this.adIframe_ = promise;
+
+    /** @private {string} */
     this.storyTemplate_ = element
       .getAttribute('data-template')
       .replace('{{ adSandbox }}', defaultIframeSandbox);
 
+    /** @private {!Promise} */
     this.iframePromise_ = untilAttached(this.element, s('.iframe'));
 
+    this.determineIframeStrategy_();
+
     // We're already keeping this value, so let's just remove the attribute to
     // clear memory.
     element.removeAttribute('data-template');
 
-    render(WrappedIframe(), this.element);
+    render(
+      WrappedIframe({
+        storyTemplate: this.storyTemplate_,
```

Comment:
Maybe just pass a `srcdoc` attr directly? and `this.useSourcedoc_ ? this.storyTemplate_ : ''`. Restricts logic to one place.

Reply:
Done.